client/src/components/RoomListCarousel.jsx

```diff
@@ -4,30 +4,55 @@ import RoomListEntry from './RoomListEntry';
 import RoomListCarouselArrow from './RoomListCarouselArrow';
 
 function RoomListCarousel(props) {
+  const displayTwoRows = (props.displayedRooms.length >= 4);
   const halfLength = Math.ceil(props.displayedRooms.length / 2);
   const firstHalf = props.displayedRooms.slice(0, halfLength);
   const secondHalf = props.displayedRooms.slice(halfLength, props.displayedRooms.length);
+
+  const roomImages = {};
+  props.displayedRooms.forEach((room) => {
+    roomImages[room.id] = [];
+  });
+
+  props.allImages.forEach((image) => {
+    if (roomImages.hasOwnProperty(image.roomId)) {
```

Comment:
airbnb eslint suggest not to use hasOwnProperty, use roomImages[image.roomId] truthiness instead